### PR TITLE
upgrade to gradle 8

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?
deprecated gradle features were being used making VRO incompatible with the latest Gradle version

Associated tickets or Slack threads:

https://github.com/orgs/department-of-veterans-affairs/projects/871/views/7?pane=issue&itemId=27157250

## How does this fix it?[^1]
upgrades gradle plugins that were using deprecated Gradle features
adds explicit dependencies where necessary
changes Gradle property names where necessary

## How to test this PR
- `./gradlew build check test`
- `./gradlew dockerComposeUp`
- `./gradlew  build -x test -x jacocoTestCoverageVerification`
[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: scripts/build.gradle [hunk 1]
```diff
@@ -2,6 +2,8 @@ plugins {
   id 'local.java.lint-shellcheck-conventions'
 }
 
+tasks.named("shellcheck").configure { dependsOn("spotlessGroovyGradle") }
```
Comment (yoomlam) [inline scripts/build.gradle:5]: Why was this line added?

**Review by yoomlam — APPROVED** [review]
Nice! 🚀

**Comment by yoomlam** [comment]
Running `./gradlew  build -x test -x jacocoTestCoverageVerification` caused error:
```
A problem was found with the configuration of task ':app:jacocoTestReport' (type 'JacocoReport').
  - Gradle detected a problem with the following location: '/Users/yoom/dev/abd-vro/app/build/resources/main'.

    Reason: Task ':app:jacocoTestReport' uses this output of task ':app:processResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

    Possible solutions:
      1. Declare task ':app:processResources' as an input of ':app:jacocoTestReport'.
      2. Declare an explicit dependency on ':app:processResources' from ':app:jacocoTestReport' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:processResources' from ':app:jacocoTestReport' using Task#mustRunAfter.

    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

```

File: scripts/build.gradle (hunk 1 above)
Comment (msnwatson) [reply scripts/build.gradle:5]: so it was actually added for the exact reason that you discovered below. there was an implicit dependency in the task in version 7, but those have been disallowed in version 8. so i'll add that to fix that error as well

Comment (yoomlam) [reply scripts/build.gradle:5]: This would be a good thing to announce at our next engineering meeting and explain how to fix, in case anyone runs a gradle task that you haven't tested.

**Review by engineer-plus-plus — APPROVED** [review]
LGTM
